docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,282 @@
+# Retrieval in CLIP-as-service
+`CLIP-as-service` offers us high-quality embeddings. Retrieval is one of the most common use cases for embeddings. Retrieval in `CLIP-as-service` support indexing a very large dataset (millions/billions) and querying within 50ms, depending on the machine.
+
+In order to implement retrieval, we add an [`AnnLite`](https://github.com/jina-ai/annlite) indexer executor after the encoder executor in CLIP-as-service.
+
+
+## Fast search in CLIP-as-service
+Indexing and searching are easy in `CLIP-as-service`:
+
+```python
+from clip_client import Client
+
+client = Client('grpc://0.0.0.0:23456')
+
+# index
+client.index(
+    [
+        Document(text='she smiled, with pain'),  # text
+        Document(uri='apple.png'),  # local image
+        Document(
+            uri='https://clip-as-service.jina.ai/_static/favicon.png'
+        ),  # online image
+    ]
+)
+
+# search
+client.search(['smile'])
+```
+
+The results will look like this:
+
+```text
+╭───────────────────────────── Documents Summary ─────────────────────────────╮
+│                                                                             │
+│   Length                 1                                                  │
+│   Homogenous Documents   True                                               │
+│   Common Attributes      ('id', 'mime_type', 'text', 'tags', 'embedding')   │
+│   Multimodal dataclass   False                                              │
+│                                                                             │
+╰─────────────────────────────────────────────────────────────────────────────╯
+╭────────────────────── Attributes Summary ───────────────────────╮
+│                                                                 │
+│   Attribute   Data type      #Unique values   Has empty value   │
+│  ─────────────────────────────────────────────────────────────  │
+│   embedding   ('ndarray',)   1                False             │
+│   id          ('str',)       1                False             │
+│   mime_type   ('str',)       1                False             │
+│   tags        ('dict',)      1                False             │
+│   text        ('str',)       1                False             │
+│                                                                 │
+╰─────────────────────────────────────────────────────────────────╯
+```
+
+You don't need to call `client.encode()` explicitly since `client.index()` will handle this for you.
```

Comment:
Is there a reason that i should expect to call `encode` at this position?

Reply:
Before you call `client.index()`, you should have embeddings first.